src/Bramus/Router/Router.php

```diff
@@ -42,10 +47,56 @@ class Router
      */
     private $serverBasePath;
 
-    /**
-     * @var string Default Controllers Namespace
-     */
-    private $namespace = '';
+	/**
+	 * @var string Default Controllers Namespace
+	 */
+	private $namespace = '';
+
+	/**
+	 * @var string The method that will take precedent over any auto determined request method
+	 */
+	private $requestMethodOverride = '';
+
+	/**
+	 * @var string The URI that will take precedent over any auto determined request URI
+	 */
+	private $requestUriOverride = '';
+
+	/**
+	 * @var bool To indicate that the URI has been requested to be overwritten, in case of an empty request URI
+	 */
+	private $requestUriOverridden = false;
+
+	/**
+	 * Allow the method of the current request to be overridden instead of being derived from the $_SERVER environment global
+	 *
+	 * @see getRequestMethodCandidate()
+	 *
+	 * @param $method The HTTP method to handle this request with
+	 *
+	 * @throws \Exception with invalid HTTP method
+	 */
+	public function setRequestMethodOverride($method)
+	{
+		$method = trim(strtoupper($method));
+
+		if(!in_array($method, explode('|', self::HTTP_METHODS))) throw new \Exception('Invalid HTTP method', 405);
+
+		$this->requestMethodOverride = $method;
+	}
+
+	/**
+	 * Allow the method of the current request to be overridden instead of being derived from the $_SERVER environment global
+	 *
+	 * @see getRequestUriCandidate()
+	 *
+	 * @param $requestUri The request URI to handle this request with
+	 */
+	public function setRequestUriOverride($requestUri)
+	{
+		$this->requestUriOverride = $requestUri;
+		$this->requestUriOverridden = true;
```

Comment:
I don't think that you need a `boolean` for that? Just check if the `requestUriOverride` isn't `null`. You could set the `requestUriOverridden` by default null. And if it's not null it means that it was overridden, even if the new `REQUEST_URI` is blank. (But it will never be null by RFC)...